openssl-dynamic/src/main/c/ssl_private.h

```diff
@@ -302,9 +302,20 @@ struct tcn_ssl_ctxt_t {
     apr_thread_rwlock_t     *mutex;
     tcn_ssl_ticket_key_t    *ticket_keys;
     unsigned int            ticket_keys_len;
+
+    /* TLS ticket key session resumption statistics */
+
+    // The client did not present a ticket and we issued a new one.
+    volatile long            ticket_keys_new;
+    // The client presented a ticket derived from the primary key
+    volatile long            ticket_keys_resume;
+    // The client presented a ticket derived from an older key, and we upgraded to the primary key.
+    volatile long            ticket_keys_renew;
+    // The client presented a ticket that did not match any key in the list.
+    volatile long            ticket_keys_fail;
 };
```

Comment:
Actually, it looks like apr has support for atomic increments....I will look into that instead.